src/main/java/com/mael/ttt/Game.java

```diff
@@ -51,7 +55,7 @@ private void setPlayers() {
     private boolean nextTurn() {
         gameUI.printBoard(board);
 
-        Player currentPlayer = (currentMark.equals("X")) ? player1 : player2;
+        Player currentPlayer = (currentMark.equals(playerMark)) ? player1 : player2;
         int index            = currentPlayer.getCellIndex(board);
         board.setCell(index, currentMark);
 
```

Comment:
Not sure I understand what you mean... `currentMark` is a field variable of `Game` that gets updated in every turn when the player has to be swapped.